fractionbounce-port.patch

```diff
@@ -64,3 +64,117 @@ index 20baa05..0e73d4a 100644
  
      def _new_ball_from_journal(self, dsobject):
          ''' Load an image from the Journal. '''
+         self._bounce_window.ball.new_ball_from_image(
+-            dsobject.file_path,
+-            os.path.join(activity.get_activity_root(), 'tmp', 'custom.png'))
++            dsobject,
++            os.path.join(os.environ['SUGAR_ACTIVITY_ROOT'], 'custom.png'))
+ 
+     def _new_background_from_journal(self, dsobject):
+         ''' Load an image from the Journal. '''
+diff --git a/ball.py b/ball.py
+index 3f96159..4de5bd8 100644
+--- a/ball.py
++++ b/ball.py
+@@ -128,22 +128,22 @@ class Ball():
+         if filename == '':
+             _logger.debug('Image file not found.')
+             return
+-        try:
```

Comment:
I have temporarily removed the try, which traces to svg_utils.py, for your reference

Reply:
Can show here

1. the content of `save_path`, and

2. the returned content of `generate_ball_svg(save_path)` ?

Reply:
@tchx84  these are the paths
```
/home/ss/.var/app/org.sugarlabs.FractionBounce/data/custom.png
```
represents save_path

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
xmlns:dc="http://purl.org/dc/elements/1.1/"
xmlns:cc="http://creativecommons.org/ns#"
xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
xmlns:svg="http://www.w3.org/2000/svg"
xmlns="http://www.w3.org/2000/svg"
xmlns:xlink="http://www.w3.org/1999/xlink"
version="1.1"
width="85"
height="120">
<image
xlink:href="file:///home/ss/.var/app/org.sugarlabs.FractionBounce/data/custom.png"
x="0"
y="35"
width="85"
height="85" />
<rect
width="85"
height="35"
ry="7.75"
x="0"
y="0"
style="fill:#ffffff;fill-opacity:1;stroke:none" />
</svg>
```
svg path

Reply:
I found a vague workaround after completely utilizing my whole day, but I want to know why this happened

Reply:
https://github.com/sugarlabs/fractionbounce/pull/15